src/coreclr/System.Private.CoreLib/src/System/Environment.CoreCLR.cs

```diff
@@ -54,12 +63,28 @@ public static partial class Environment
         //    IP for bucketing. If the exception object is not preallocated, it will use the bucket
         //    details contained in the object (if any).
         [DoesNotReturn]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void FailFast(string? message, Exception? exception);
+        [DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
+        public static void FailFast(string? message, Exception? exception)
+        {
+            StackCrawlMark mark = StackCrawlMark.LookForMyCaller;
+            FailFast(new StackCrawlMarkHandle(ref mark), new StringHandleOnStack(ref message), ObjectHandleOnStack.Create(ref exception));
+        }
+
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Environment_FailFastWithException")]
+        [DoesNotReturn]
+        private static partial void FailFast(StackCrawlMarkHandle mark, StringHandleOnStack message, ObjectHandleOnStack exception);
```

Comment:
We can have just one QCall overload to implement all public entrypoints (pass null on the managed side instead of the native side).

Reply:
I have consolidated to a single QCall and removed the comments admonishing the reader to not do that.